### DISCUSSION TITLE
code-server not loading on iPad

### DISCUSSION THREAD

**Discussion by Freddy-0** [coder/code-server#3619] [Q&A] [answered]
## OS/Web Information

- Web Browser: safari
- Local OS: ipad os 1.14.6
- Remote OS: unraid
- `code-server --version`: 1.10.2

## Steps to Reproduce

installed code-server on Unraid thru Community applications, image based of https://hub.docker.com/r/linuxserver/code-server/tags?page=1&ordering=last_updated. configured everithyng. Using Nginx proxy manager with authelia and cloudflare. everything works fine on desktop, But on ipad in every browser there is just a blank screen. 

## Expected

vs-studio opens after login

## Actual

just a white screen

## Logs

unfortunately i cant collect logs from ipad, because i dont have a mac to read them out

This issue can be reproduced in VS Code: Yes/No

**Comment by jsjoeio** [comment]
What happens if you refresh/reload the page?

**Comment by Freddy-0** [comment]
When refreshing / using command+r on safari i still get the white screen

**Comment by jsjoeio** [comment]
Hmm... can you [record a video](https://support.apple.com/guide/ipad/take-a-screenshot-or-screen-recording-ipad08a40f3b/ipados) and show me the whole process on your iPad? (I just tested on my iPad and things are working as expected)

**Comment by Freddy-0** [comment]
on local ip it works fine but with the domain it wont load

**Comment by jsjoeio** [comment]
Hmm... this appears to be a local environment issue (at least, I say this because things are working on my setup with Railway). I'm going to convert this to a Discussion and we can continue the conversation there.

**Reply by jsjoeio** [reply]
It's very difficult to tell what could be causing this without at least seeing the browser logs 🤔 You say it loads fine on Desktop?

I wonder if it's a Safari issue. It's too bad you don't have a way to test that on Desktop with Safari because then we could look at the logs. The only way I know how to look at browser logs on iPad is with [this app](https://apps.apple.com/us/app/inspect-browser/id1203594958) but it costs $6.99.

## Screenshot

My setup (this was recorded today for a YouTube video)

https://user-images.githubusercontent.com/3806031/122278152-77cb5880-ce9b-11eb-8a14-3b0f04012cd0.MP4

**Reply by teja-self-taught-cs** [reply]
I'm having issues installing code-server in iPad using raspberry pi 4 which is connected to my wifi using ethernet cable, please help I'm learning full stack web developement and want to code on my iPad..
![IMG_20221102_131112](https://user-images.githubusercontent.com/104780299/199429735-bb4d3df1-e6d7-4a27-8aa6-bca6ca7b38de.jpg)

**Reply by jsjoeio** [reply]
please use npm!

**Comment by Freddy-0** [comment]
just some more basic expalnation on my setup:
Cloudflare just proxies the cname thru argo tunnel to my network. after that i set up an Nginx Proxy manager instance. which is configured as follow:
![image](https://user-images.githubusercontent.com/72572280/122278015-cc47f580-cee6-11eb-914e-af9b222ef5df.png)
with the following config added for authelia: location /authelia {
internal;
set $upstream_authelia http://192.168.178.104:9091/api/verify;
proxy_pass_request_body off;
proxy_pass $upstream_authelia;    
proxy_set_header Content-Length "";

# Timeout if the real server is dead
proxy_next_upstream error timeout invalid_header http_500 http_502 http_503;
client_body_buffer_size 128k;
proxy_set_header Host $host;
proxy_set_header X-Original-URL $scheme://$http_host$request_uri;
proxy_set_header X-Real-IP $remote_addr;
proxy_set_header X-Forwarded-For $remote_addr; 
proxy_set_header X-Forwarded-Proto $scheme;
proxy_set_header X-Forwarded-Host $http_host;
proxy_set_header X-Forwarded-Uri $request_uri;
proxy_set_header X-Forwarded-Ssl on;
proxy_redirect  http://  $scheme://;
proxy_http_version 1.1;
proxy_set_header Connection "";
proxy_cache_bypass $cookie_session;
proxy_no_cache $cookie_session;
proxy_buffers 4 32k;

send_timeout 5m;
proxy_read_timeout 240;
proxy_send_timeout 240;
proxy_connect_timeout 240;
}

location / {
set $upstream_Service $forward_scheme://$server:$port;
proxy_pass $upstream_Service;

auth_request /authelia;
auth_request_set $target_url https://$http_host$request_uri;
auth_request_set $user $upstream_http_remote_user;
auth_request_set $groups $upstream_http_remote_groups;
proxy_set_header Remote-User $user;
proxy_set_header Remote-Groups $groups;
error_page 401 =302 https://auth.higgs-boson.eu/?rd=$target_url;

client_body_buffer_size 128k;

proxy_next_upstream error timeout invalid_header http_500 http_502 http_503;

send_timeout 5m;
proxy_read_timeout 360;
proxy_send_timeout 360;
proxy_connect_timeout 360;

proxy_set_header Host $host;
proxy_set_header Upgrade $http_upgrade;
proxy_set_header Connection upgrade;
proxy_set_header Accept-Encoding gzip;
proxy_set_header X-Real-IP $remote_addr;
proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
proxy_set_header X-Forwarded-Proto $scheme;
proxy_set_header X-Forwarded-Host $http_host;
proxy_set_header X-Forwarded-Uri $request_uri;
proxy_set_header X-Forwarded-Ssl on;
proxy_redirect  http://  $scheme://;
proxy_http_version 1.1;
proxy_set_header Connection "";
proxy_cache_bypass $cookie_session;
proxy_no_cache $cookie_session;
proxy_buffers 64 256k;

set_real_ip_from 192.168.1.0/16;
real_ip_header X-Forwarded-For;
real_ip_recursive on;

}

**Reply by jsjoeio** [reply]
Hmm...I have less experience with proxies, ngnix, etc. Maybe @code-asher sees something though?

**Reply by Freddy-0** [reply]
what gets me caught, is that everything except ipad /iphone works fine (only apple devices i can test rn)i

**Reply by Freddy-0** [reply]
(maybe i can get acess to an imac,, so i can read the logs tomorrow)

**Reply by Freddy-0** [reply]
so, just remembered i have an old mac os vm installed, i get the error "1006 websockets close "

**Comment by Freddy-0** [comment, marked as answer]
Here is the full error: 

Error: WebSocket close with status code 1006

R@https://vs.higgs-boson.eu/static/387b12ef4ca404ffd39d84834e1f0776e9e3c005/usr/local/share/.config/yarn/global/node_modules/code-server/lib/vscode/out/vs/workbench/workbench.web.api.js:634:2136381
https://vs.higgs-boson.eu/static/387b12ef4ca404ffd39d84834e1f0776e9e3c005/usr/local/share/.config/yarn/global/node_modules/code-server/lib/vscode/out/vs/workbench/workbench.web.api.js:634:2138140
Error: {}

**Reply by Freddy-0** [reply]
how could i fix that?

**Reply by jsjoeio** [reply]
Woo! Now we're getting somewhere. Hmm... The WebSocket closed 🤔 I feel like I've seen @code-asher help someone with this. Let me do some investigating.

**Reply by jsjoeio** [reply]
I found [this](https://github.com/cdr/code-server/discussions/2698#discussioncomment-359372): sounds like the same error.

The docs link is old. [Here](https://github.com/cdr/code-server/blob/main/docs/guide.md#nginx) is the newest one for working with NGINX 

Again, I haven't used a setup like this with NGINX but I wonder if it has to do with `proxy_pass` in your config?

Similar issues in the past:
- https://github.com/cdr/code-server/issues/2491

**Reply by Freddy-0** [reply]
i dont know if that did the trick, because i tried a lot of stuff, but after changing cloudflare to use flexible encryption and usig this nginx config:

 # ------------------------------------------------------------
# +Domain+
# ------------------------------------------------------------


server {
  set $forward_scheme http;
  set $server         "192.168.178.104";
  set $port           8500;

  listen 8080;
listen [::]:8080;

listen 4443 ssl http2;
listen [::]:4443;


  server_name +domain+;


  # Let's Encrypt SSL
  include conf.d/include/letsencrypt-acme-challenge.conf;
  include conf.d/include/ssl-ciphers.conf;
  ssl_certificate /etc/letsencrypt/live/npm-5/fullchain.pem;
  ssl_certificate_key /etc/letsencrypt/live/npm-5/privkey.pem;




# Asset Caching
  include conf.d/include/assets.conf;


  # Block Exploits
  include conf.d/include/block-exploits.conf;








proxy_set_header Upgrade $http_upgrade;
proxy_set_header Connection $http_connection;
proxy_http_version 1.1;



  access_log /data/logs/proxy_host-33.log proxy;







  location / {

    





    
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection $http_connection;
    proxy_http_version 1.1;
    

    # Proxy!
    include conf.d/include/proxy.conf;
  }


  # Custom
  include /data/nginx/custom/server_proxy[.]conf;
}

it magically worked and didnt return the error 1006 

but i stil cant realy address the root cause of the problem, i wil try to break it again and see what does the trick

**Comment by Freddy-0** [comment]
So, after reverting cloudflare back to strict and redoing my nginx entry like i had it before it still works. unfortunately nothing else comes to mind that could have influenced the result

**Reply by jsjoeio** [reply]
That's so strange! I have no idea what it could be. Regardless, I'm glad it's working now 😄 Go team!

**Comment by alexaptib** [comment]
Hi @jsjoeio, I am having the same issue when accessing through an iPad, using a getumbrel node. I get to the login page and input the password, but then there is only a white screen. I can access it with no problem using my macbook and iPhone though.

I tried using chrome and safari, both don't work.

Any ideas about what it going on?

**Reply by jsjoeio** [reply]
What happens if you hit Cmd + R on iPad at the white screen? That refreshes the page.

> I tried using chrome and safari, both don't work.

Yeah, on iPad, they use the same engine under the hood so you won't see much difference between the two.

If you can get access to the iPad browser logs when the white screen happens, that would help!

- https://apps.apple.com/us/app/inspect-browser/id1203594958
- https://developer.apple.com/safari/tools/
- https://www.itsupportguides.com/knowledge-base/ios/how-to-debug-safari-on-ipad-ipod-iphone/

The other question is if you're accessing code-server via HTTPS on iPad? If not, that may cause issues.
